src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -425,6 +430,19 @@ protected void checkImageWithAutoPull(ServiceHub hub, String image, String regis
         }
     }
 
+    private synchronized Set<String> getPreviouslyPulledImageCache() {
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Set<String> cache = (Set) session.getUserProperties().get(CONTEXT_KEY_PREVIOUSLY_PULLED);
+        if (cache == null) {
+            System.out.println("creating new context");
+
+            cache = Sets.newConcurrentHashSet();
```

Comment:
Funnily this google code is not much more than a 

```
Collections.newSetFromMap(new ConcurrentHashMap<String,Boolean>())
```

@jgangemi At the moment when testing this I have problems with the FailSafePlugin which tries to write the `seesion.getUserProperties()` to disk because of a String class cast exception. And as Properties are supposed to carry String values this is not so a big surprise.

What do you think, should we serialize the map to JSON string, or do you can think about a different sharing mechanism among multiple reactor projects ?